registry/repository.go

```diff
@@ -143,5 +218,75 @@ func (c *repositoryClient) uploadSignatureManifest(ctx context.Context, subject,
 		PackImageManifest:   c.OCIImageManifest,
 	}
 
-	return oras.Pack(ctx, c.Repository, ArtifactTypeNotation, []ocispec.Descriptor{blobDesc}, opts)
+	return oras.Pack(ctx, c.Target, ArtifactTypeNotation, []ocispec.Descriptor{blobDesc}, opts)
+}
+
+// referrers returns referrer nodes of desc in target and filter by artifactType
+func referrers(ctx context.Context, target content.ReadOnlyGraphStorage, desc ocispec.Descriptor, artifactType string) ([]ocispec.Descriptor, error) {
```

Comment:
In the current implementation of notation, the `artifactType` is fixed. Thus, we can hard code it.